api/http/saml.go

```diff
@@ -168,39 +150,46 @@ func (service SamlResponseHandler) serveAuthnResponse(encodedResponse string, w
 		redirectAccessDenied(w, r)
 	}
 
-	// Generate jwt token
-	signedToken, _, err := service.Token.NewToken(account.ID, sessionIndex, api.SingleSignOnAudience)
-	if err != nil {
-		service.Log.WarnError(api.JWTError, err, api.LogFields{"account": account})
-		redirectAccessDenied(w, r)
+	sessionKey, authErr := service.Session.UserDidAuthenticate(account.ID, simplestore.NonNullString(sessionIndex))
+	if authErr != nil {
+		service.Log.WarnError("bad session get", authErr, api.LogFields{"account": account.ID})
+		RespondWithStructuredError(w, "bad session get", http.StatusInternalServerError)
 		return
 	}
 
-	service.Log.Info(api.SamlValid, api.LogFields{"account": account})
-	if cookieDomain == "" {
-		service.Log.Warn(api.CookieDomainNotSet, api.LogFields{})
-		// Default to frontend host
-		uri, _ := url.Parse(redirectTo)
-		cookieDomain = strings.Split(uri.Host, ":")[0]
-	}
-	expiration := time.Now().Add(time.Duration(1) * time.Minute)
-	cookie := &http.Cookie{
-		Domain:   cookieDomain,
-		Name:     "token",
-		Value:    signedToken,
-		HttpOnly: false,
-		Path:     "/",
-		MaxAge:   60,
-		Expires:  expiration,
-	}
-	http.SetCookie(w, cookie)
+	service.Cookie.AddSessionKeyToResponse(w, sessionKey)
+
 	http.Redirect(w, r, redirectTo, http.StatusFound)
 }
 
 func (service SamlResponseHandler) serveLogoutResponse(encodedResponse string, w http.ResponseWriter, r *http.Request) {
+	// TODO: validate the SAML Logout Response and tie it to a session via the SLO request ID
```

Comment:
Can you clarify what you mean by that?

Reply:
I'll expand on the TODO in the comments and make an issue for it. As is, we are not validating the logout response from WSO2. We should be:

1. verifying that the signature is correct
2. checking that the ID in the response is the same as the one we made the logout request for. This will confirm that the logout message was for the intended user.

1. is more important than 2, for sure.